pkg/client/connector/k8s_config.go

```diff
@@ -1,17 +1,47 @@
 package connector
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
 	"sort"
 
 	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/rest"
 
 	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/telepresenceio/telepresence/v2/pkg/iputil"
 )
 
+// The dns config part of the kubeconfigExtension that is read from the kubeconfig
+type dns struct {
+	// LocalIP is the address of the local DNS server. This entry is only
+	// used on Linux system that are not configured to use systemd.resolved and
+	// can be overridden by using the option --dns on the command line and defaults
+	// to the first line of /etc/resolv.conf
+	LocalIP iputil.IPKey `json:"local-ip,omitempty"`
+
+	// RemoteIP is the address of the cluster's DNS service. It will default
+	// to the IP of the kube-dns.kube-system or the dns-default.openshift-dns service.
+	RemoteIP iputil.IPKey `json:"remote-ip,omitempty"`
+
+	// ExcludeSuffixes are suffixes for which the DNS resolver will always fail (or fallback
+	// in case of the overriding resolver).
+	ExcludeSuffixes []string `json:"exclude-suffixes,omitempty"`
+
+	// IncludeSuffixes are suffixes for which the DNS resolver will always attempt to do
+	// a lookup. Includes have higher priority than excludes.
+	IncludeSuffixes []string `json:"include-suffixes,omitempty"`
+}
+
+// kubeconfigExtension is an extension read from the selected kubeconfig Cluster.
```

Comment:
Is the reasoning for putting this configuration in the kubeconfig because users could potentially have different values for different clusters?  

I guess I'm torn because I'd prefer for all of our config to be in one place (e.g. `telepresence/config.yml`) because once configuration is spread across multiple files, I feel like it can get confusing to figure out what you configure where.  

I think I still need to do some more thinking about this, but just wanted to check + ensure I understood the reasoning for this decision.

Reply:
I had this in the `config.yml` initially. Then @LukeShu suggested that we should use the kubeconfig extension mechanism for the `also-proxy` stuff because it's bound to be cluster specific and I really liked that idea. Since the DNS resolver resides in the cluster nowadays, it's also cluster specific (besides the `local-ip`, but separating that out seemed worse than keeping the config together). The include-suffixes for instance, is partly aimed at solving the problem with pods using custom DNS-resolvers.

Reply:
Ah okay that makes a lot of sense. I was on the fence initially, but after thinking more about it and listening to our reasoning, I think it makes more sense for cluster-specific config to be added via a kubeconfig extension and the `config.yml` can be for more "global" telepresence config. I think as long as we document that well, then it shouldn't be confusing for our users.